tools/rum-models-generator/Sources/RUMModelsGeneratorCore/Input/JSON/JSONSchemaToJSONTypeTransformer.swift

```diff
@@ -33,12 +37,17 @@ internal class JSONSchemaToJSONTypeTransformer {
     // MARK: - Transforming ambiguous types
 
     private func transformSchemaToAnyType(_ schema: JSONSchema, named name: String) throws -> JSONType {
-        // RUMM-2022: Pick first schema of OneOf to workaround change introduced in
+        context.enter((name, schema))
+        defer { context.leave() }
+
+        // RUMM-2022: Pick first schema of OneOf to workaround `action.id` change introduced in
         // https://github.com/DataDog/rum-events-format/pull/57
         //
         // Supporting multiple types for single property need to be addressed
-        if let oneOf = schema.oneOf?.first {
-            return try transformSchemaToAnyType(oneOf, named: name)
+        if context.parent?.name == "action" && context.current?.name == "id" {
+            if let oneOf = schema.oneOf?.first {
+                return try transformSchemaToAnyType(oneOf, named: name)
+            }
         }
```

Comment:
Here, I make the `action.id` patch to be more strict and only applied where we know it should be applied. This is to avoid overlooking some other changes that might went through unnoticed.